src/test/java/gmailgdogra/ExtractOfficersServiceTest.java

```diff
@@ -3,60 +3,35 @@
 import gmailgdogra.pojo.Officer;
 import gmailgdogra.pojo.SwipeRecord;
 import gmailgdogra.service.ExtractOfficersService;
-import gmailgdogra.service.ReadXlsxService;
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
 
-import java.io.FileInputStream;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
-@SpringBootTest
 class ExtractOfficersServiceTest {
 
-    private final ReadXlsxService readXlsxService;
+    private final Officer officerOne = new Officer("Officer", "One");
+    private final Officer officerTwo = new Officer("Officer", "Two");
 
-    @Autowired
-    public ExtractOfficersServiceTest(ReadXlsxService readXlsxService) {
-        this.readXlsxService = readXlsxService;
-    }
+    private final Set<Officer> expected = new HashSet<>(Arrays.asList(
+            officerTwo,
+            officerOne
+            ));
 
-    private final Set<Officer> expectedOfficers = new HashSet<>(Arrays.asList(
-            new Officer("Officer", "One"),
-            new Officer("Officer", "Two"),
-            new Officer("Officer", "Three"),
-            new Officer("Officer", "Four"),
-            new Officer("Officer", "Five"),
-            new Officer("Officer", "Six"),
-            new Officer("Officer", "Seven"),
-            new Officer("Officer", "Eight"),
-            new Officer("Officer", "Nine"),
-            new Officer("Officer", "Ten"),
-            new Officer("Officer", "Eleven"),
-            new Officer("Officer", "Twelve")
-    ));
+    private final List<SwipeRecord> inputList = Arrays.asList(
+            new SwipeRecord(officerOne, null, null),
+            new SwipeRecord(officerOne, null, null),
+            new SwipeRecord(officerTwo, null, null)
+    );
 
     @Test
-    void test_from() {
-        Path path = Paths.get("src/main/resources/test file 1.xlsx");
-        Set<Officer> officers = null;
-
-        try {
-            List<SwipeRecord> swipeRecords = readXlsxService.readAllRows(new FileInputStream(path.toFile()));
-            officers = ExtractOfficersService.from(swipeRecords);
-        } catch (Exception e) {
-            System.out.println(e.getMessage());
-            Assertions.fail("Unable to read file at " + path);
-        }
-
-        assertTrue(officers.containsAll(expectedOfficers));
+    public void test_distinctReturn() {
+        assertEquals(expected, ExtractOfficersService.from(inputList));
     }
-}
\ No newline at end of file
+
```

Comment:
If you use JUnit5, you can nicely arrange your tests using @Nested inner classes and @DisplayName to make test names be like "Given empty input, when someMethod(), then return false" and so on. The more tests you have, the more important their structure is.